### ISSUE TITLE
Remove dependency to ndk for sentry-android by default

### ISSUE THREAD

**Issue by narenkmanoharan** [getsentry/sentry-android#441]
_Platform:_
- [x]  Android -> If yes, which Device API (and compileSdkVersion/targetSdkVersion/Build tools) version? 29 - compileSdkVersion - 29 - targetSdkVersion - 29
- [ ]  Java -> If yes, which Java (and sourceCompatibility/targetCompatibility) version?
- [ ]  Kotlin -> If yes, which Kotlin (and jvmTarget) version?
- [ ]  NDK -> If yes, which NDK/CMake version?
- [ ]  React-Native -> If yes, which version?

_IDE:_
- [x]  Android Studio -> If yes, which version? Android Studio 3.5.3 (Build #AI-191.8026.42.35.6010548)
- [ ]  IntelliJ -> If yes, which version?
- [ ]  Other -> If yes, which one?

_Build system:_
- [X]  Gradle -> If yes, which version? 6.4.1
- [ ]  Buck -> If yes, which version?
- [ ]  Bazel -> If yes, which version?
- [ ]  Other -> If yes, which one?

_Android Gradle Plugin:_
- [x]  Yes -> If yes, which version? 3.5.3
- [ ]  No

_Sentry Android Gradle Plugin:_
- [ ]  Yes -> If yes, which version?
- [x]  No

_Proguard/R8:_
- [ ]  Enabled
- [x]  Disabled

_sentry-android installed with:_
- [ ] JCenter
- [ ] Bintray
- [x] Maven Central
- [ ] Manually

The version of sentry-android:
**2.1.2**

---
I have the following issue:

We recently updated from sentry-android v1.7.27 to sentry-android v2.1.2. We don't have a native libraries included in our apk but looks like the sentry-android 2.0 and above has a dependency to a native library. And this means that we would be shipping native code as well. We would prefer not to do this. Would it be possible to make it optional to include the ndk library with the default sentry-android dependency?

![image](https://user-images.githubusercontent.com/9508432/83293076-1527f180-a1a0-11ea-8237-45bbceae6fee.png)

Thanks!

**Comment by marandaneto** [comment]
hey @narenkmanoharan thanks for asking.

https://docs.sentry.io/platforms/android/#advanced-usage

TLDR:
just use `io.sentry:sentry-android-core` instead of `io.sentry:sentry-android`

**Comment by narenkmanoharan** [comment]
Thanks!

**Comment by edenman** [comment]
I just hit this too.  IMO this should really be added to the readme, adding an implicit NDK dependency is crazy.

**Comment by marandaneto** [comment]
@edenman this is written in our official docs and the docs are linked in the readme.

**Comment by edenman** [comment]
Ah yes, there it is, 3/4 down the incredibly long page in the "advanced usage" section.  

I'd argue it should go very prominently in the "Integrating the SDK" section.  Adding an NDK dependency is not something I think most people would think to even look for, at least I sure didn't.

**Comment by marandaneto** [comment]
@edenman while I agree with your statement, it's a very personal question.

some people prefer to have everything enabled and to have an enriched even, but they can opt-out from features, while other people prefer a lighter version of it, but still being able to opt-in from features.

the thing is, most of the people don't read the docs and they end up with the lighter version and never get to know about the opt-in features and this leads to issues like -> "why am I not getting NDK events?" way more than the ones like -> "why am I getting NDK events?" so we've decided to use the opt-out approach, just sharing our experience :)
